godtools/App/Features/AppLanguage/Data-DomainInterface/GetDownloadableLanguagesListRepository.swift

```diff
@@ -0,0 +1,100 @@
+//
+//  GetDownloadableLanguagesListRepository.swift
+//  godtools
+//
+//  Created by Rachael Skeath on 12/15/23.
+//  Copyright © 2023 Cru. All rights reserved.
+//
+
+import Foundation
+import Combine
+
+class GetDownloadableLanguagesListRepository: GetDownloadableLanguagesListRepositoryInterface {
+    
+    private let languagesRepository: LanguagesRepository
+    private let downloadedLanguagesRepository: DownloadedLanguagesRepository
+    private let getTranslatedLanguageName: GetTranslatedLanguageName
+    private let resourcesRepository: ResourcesRepository
+    private let localizationServices: LocalizationServices
+    
+    init(languagesRepository: LanguagesRepository, downloadedLanguagesRepository: DownloadedLanguagesRepository, getTranslatedLanguageName: GetTranslatedLanguageName, resourcesRepository: ResourcesRepository, localizationServices: LocalizationServices) {
+        
+        self.languagesRepository = languagesRepository
+        self.downloadedLanguagesRepository = downloadedLanguagesRepository
+        self.getTranslatedLanguageName = getTranslatedLanguageName
+        self.resourcesRepository = resourcesRepository
+        self.localizationServices = localizationServices
+    }
+    
+    func getDownloadableLanguagesPublisher(currentAppLanguage: AppLanguageDomainModel) -> AnyPublisher<[DownloadableLanguageListItemDomainModel], Never> {
+        
+        return Publishers.CombineLatest(
+            languagesRepository.getLanguagesChanged(),
+            downloadedLanguagesRepository.getDownloadedLanguagesChangedPublisher()
+        )
+        .map { _ in
+            
+            return self.languagesRepository.getLanguages().map { language in
+                                
+                let languageNameInOwnLanguage = self.getTranslatedLanguageName.getLanguageName(
+                    language: language,
+                    translatedInLanguage: language.id
+                )
+                let languageNameInAppLanguage = self.getTranslatedLanguageName.getLanguageName(
+                    language: language,
+                    translatedInLanguage: currentAppLanguage
+                )
+                
+                let toolsAvailableText = self.getToolsAvailableText(for: language.languageCode, translatedIn: currentAppLanguage)
+                
+                let downloadStatus = self.getDownloadStatus(for: language.id)
+                
+                return DownloadableLanguageListItemDomainModel(
+                    languageId: language.id,
+                    languageNameInOwnLanguage: languageNameInOwnLanguage,
+                    languageNameInAppLanguage: languageNameInAppLanguage,
+                    toolsAvailableText: toolsAvailableText,
+                    downloadStatus: downloadStatus
+                )
+            }
+        }
+        .eraseToAnyPublisher()
+    }
+    
+    func observeLanguagesChangedPublisher() -> AnyPublisher<Void, Never> {
```

Comment:
Hey @rachaelblue go ahead and exclude this exposed observer.  Now that we're pushing business logic to the interface implementation we'll leave it up to the implementation to observer data layer changes.  

Eventually I'm going to go through and remove any observers on the domain layer interfaces and push those to the data layer implementation.  This is a result of our last architecture discussion.

Reply:
Good point!  I like having the observer in the interface implementation with the rest of the logic, too.